source/math/FloatTransform2D.ooc

```diff
@@ -68,6 +68,10 @@ FloatTransform2D: cover {
 	rotationZ ::= this b atan2(this a)
 	inverse: This { get {
 		determinant := this determinant
+		version (safe) {
+			if (determinant == 0)
+				raise("Determinant is zero in FloatTransform2D inverse()")
```

Comment:
This check needed adding here, yes, but I'm not sure it belongs in a `safe` block. `safe` blocks are used for one reason: performance. Bounds checking every single indexing of an array or collection means lots of unnecessary work, because we should never access out of bounds, and when we do, it's because we made a mistake somewhere, and then the bounds check can help us figure out where, so we can fix it. But when we want to compute the inverse of a transform, and the determinant is 0, something is wrong with the transform, and there is no way we can know this unless we compute the determinant (as far as I know). When this happens, it should not be hidden from us inside a `safe` block. Does this make sense?